index.js

```diff
@@ -147,8 +147,8 @@ module.exports = function (source) {
 
     const template = fm
       .html
-      .replace(/<code(\s[^>]+)>(.+)<\/code>/sg, "<code$1 dangerouslySetInnerHTML={{ __html: `$2` }} />")
-      .replace(/<code>(.+)<\/code>/g, "<code dangerouslySetInnerHTML={{ __html: `$1` }} />");
+      .replace(/<code(\s[^>]+)>(.+?)<\/code>/sg, "<code$1>{`$2`}</code>")
+      .replace(/<code>(.+?)<\/code>/g, "<code>{`$1`}</code>");
```

Comment:
I intended to use `dangerouslySetInnerHTML` over just preserving with `` {`innerContent`} ``, because `<code>` may have HTML for some reasons like syntax highlighting (For instance, [Prism](https://prismjs.com/) produces many `<span>` under `<code>`).

so

```suggestion
      .replace(/<code(\s[^>]+)>(.+?)<\/code>/sg, "<code$1 dangerouslySetInnerHTML={{ __html: `$2`}} />")
      .replace(/<code>(.+?)<\/code>/g, "<code dangerouslySetInnerHTML={{ __html: `$1`}} />");
```

works for your case as well I hope.

Reply:
Many thanks for your prompt feedback!

`dangerouslySetInnerHTML` sounds dangerous :sweat_smile:
Are you sure that add value to some current user of your loader? 

I gave it a quick try. Given `<span>TEST</span>` in a `<code />` section, my version generates `&lt;span&gt;TEST&lt;/span&gt;` and your `<span>TEST</span>` not `TEST`.

I am not in favor in adding your suggestion unless I am missing soemthing elst as it adds some complexity without having a major benefit.

Reply:
> sounds dangerous 😅

Indeed. Originally I avoid using that until I receive your issue a couple of weeks ago.
But those contents are completely passed from controlled sources (unlike CGM), so it'd be acceptable.

> my version generates &lt;span&gt;TEST&lt;/span&gt; and your <span>TEST</span> not TEST.

I guess you tried to write `<span>` in the code block section directly?
`markdown-it-prism` adds `<span>` in markdown compilation process, so need to use that reproduce the same situation (sorry, my explanation was insufficient).

As [the doc already giving a sample usage of markdown-it-prism](https://hmsk.github.io/frontmatter-markdown-loader/options.html#configure-markdown-it), intending to cover. And saw some users who use (for Vue though)

Reply:
Thanks for your clarification.

Reply:
Thanks for your generosity!